transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/Defaults.java

```diff
@@ -97,16 +99,31 @@ private static final String getVersion(final String platform) {
               DependencyConfiguration.builder(RUNTIME_ONLY, "com.linkedin.transport:transportable-udfs-test-hive", TRANSPORT_VERSION).build()
           ),
           ImmutableList.of(new ShadedJarPackaging(ImmutableList.of("org.apache.hadoop", "org.apache.hive"), null))),
-      new Platform(SPARK,
+      new Platform(SPARK_2_11,
           Language.SCALA,
           SparkWrapperGenerator.class,
           JavaLanguageVersion.of(8),
           ImmutableList.of(
-              DependencyConfiguration.builder(IMPLEMENTATION, "com.linkedin.transport:transportable-udfs-spark", TRANSPORT_VERSION).build(),
-              DependencyConfiguration.builder(COMPILE_ONLY, "org.apache.spark:spark-sql_2.11", SPARK_VERSION).build()
+              DependencyConfiguration.builder(IMPLEMENTATION, "com.linkedin.transport:transportable-udfs-spark_2.11", TRANSPORT_VERSION).build(),
+              DependencyConfiguration.builder(COMPILE_ONLY, "org.apache.spark:spark-sql_2.11", SPARK_2_11_VERSION).build()
           ),
           ImmutableList.of(
-              DependencyConfiguration.builder(RUNTIME_ONLY, "com.linkedin.transport:transportable-udfs-test-spark", TRANSPORT_VERSION).build()
+              DependencyConfiguration.builder(RUNTIME_ONLY, "com.linkedin.transport:transportable-udfs-test-spark_2.11", TRANSPORT_VERSION).build()
+          ),
+          ImmutableList.of(new ShadedJarPackaging(
+              ImmutableList.of("org.apache.hadoop", "org.apache.spark"),
+              ImmutableList.of("com.linkedin.transport.spark.**")))
+          ),
+      new Platform(SPARK_2_12,
```

Comment:
Is there a way to just create new `platforms` here (i.e. similar to what you are doing in this class) but maintain one Spark physical module (as opposed to `spark_2.11` and `spark_2.12`), and manipulate the Spark platforms using `com.linkedin.transport.plugin.TransportPlugin#configurePlatform`?

Reply:
We would still need to publish two physical modules: transportable-udfs-spark_2.11 and transportable-udfs-spark_2.12 because the plugin artifact (ie.  com.linkedin.transport:transportable-udfs-plugin:+) depends on it transitively. 
We shouldnt modify the platforms here to create  transportable-udfs-spark_2.11, _2.12 at user's build time. Not sure if it's possible to do that.